llpc/context/llpcCompiler.cpp

```diff
@@ -743,13 +750,13 @@ Result Compiler::buildPipelineWithRelocatableElf(Context *context, ArrayRef<cons
       auto pipelineInfo = reinterpret_cast<const GraphicsPipelineBuildInfo *>(context->getPipelineBuildInfo());
       cacheHash = PipelineDumper::generateHashForGraphicsPipeline(pipelineInfo, true, true, stage);
 #if LLPC_CLIENT_INTERFACE_MAJOR_VERSION < 38 || LLPC_ENABLE_SHADER_CACHE
-      userShaderCache = pipelineInfo->shaderCache;
+      userShaderCache = static_cast<IShaderCache *>(static_cast<void *>(pipelineInfo->shaderCache));
```

Comment:
Why is this needed, here and elsewhere?

Reply:
The forward declaration was moved into Vkgc, while the class/struct definition remained in Llpc. I think this was an innocuous breakage that nobody noticed since this code is not currently being used with the current llpc client interface. I suggest a proper fix can follow once this code is built and tested.

Reply:
Ah, one more argument against preprocessor use... I think it can be forgiven for now ;)